src/cephadm/cephadm

```diff
@@ -1499,6 +1695,9 @@ def command_deploy():
     # type: () -> None
     (daemon_type, daemon_id) = args.name.split('.', 1)
 
+    l = FileLock(args.fsid + '.deploy')
```

Comment:
add `<service>.<id>` to the filename? .. to allow deploying OSDs etc in parallel ..

Reply:
well - actually we shouldn't allow deploying osds in parallel at all. No idea what happens, if we deploy osds using two different drive groups on one system concurrently. 

On the other hand, we should allow running `device ls` and `service ls` and deploying (osd|mon|mgr|...)s in parallel

Reply:
I think we should stuck to a global deploy lock for now.  otherwise, we need to introduce read/write shared locks, since you want to allow >1 mds/rgw/wahtever deploys to proceed, but those should still conflict with (say) an rm-cluster.  I don't think it's worth the complexity.  The most important goal right now is to get parallelism across multiple hosts; less important on a single host--we can always improve that later.